test/utils/fixture/fixture.go

```diff
@@ -212,10 +215,27 @@ func getTeardownTimeFromEnvVar() time.Duration {
 	return val
 }
 
+func getResourcesCooldownThresholdFromEnvVar() int {
+	raw, ok := os.LookupEnv("E2E_NROP_RESOURCES_COOLDOWN_THRESHOLD")
+	if !ok {
+		return defaultResCooldownThreshold
+	}
+	val, err := strconv.Atoi(raw)
+	if err != nil {
+		klog.Errorf("cannot parse the provided test cooldown threshold for resources (fallback to default: %v): %v", defaultResCooldownThreshold, err)
+		return defaultResCooldownThreshold
+	}
+	return val
+}
+
 func GetCoolDownTime() time.Duration {
 	return cooldownTime
 }
 
 func GetTeardownTime() time.Duration {
 	return teardownTime
 }
+
+func GetResourcesCooldownThreshold() int {
+	return resCooldownThreshold
+}
```

Comment:
why do we need this getter?

Reply:
because we want to protect this from manipulations by other parts of the code, and to retain its original value we got from the env var.

Reply:
Adding a getter is indeed a way (pretty much the only way in golang) to enforce the fact the client code should access the value in read-only and not change it. The change is, of course, done intentionally (there are no accidents in code) but is missed during the review or the testing or both and causes an unwanted mutation.

However adding a getter fixes the problem, but increases the API surface (functions AND exposed variables are part of the API! the former is better than the latter). A even better fix is not increasing the API surface, and incresae cohesion while reducing coupling.

We can call "cohesion" how much the related code is logically close together. Accessing a global variable, for example, has likely low cohesion because references to the variable can be spread all across the codebase.
Accessing a local private member has higher cohesion, because all the code has to be confined in the same package.

We can call "coupling" how much two pieces of code depend on each other. A package coupled to another, perhaps because of bad API, need to change when the other package change, because for example it depends on internal behavior and not on the interface, or because the interface is too big.

Having a getter can _potentially_ decrease cohesion and (by definition, because it is adding to the API surface) always increase coupling.
There's a possible way, I believe, to increse cohesion _AND_ reduce coupling if we change `Cooldown` a bit, and I'm suggesting that in another comment.